### PR TITLE
parser: `mut:` interface methods, mut interface fn parameter

### LINKED ISSUE #1081: It is possible to mutate an immutable value through an interface
**V version:0.1.13** 
**OS:linux** 

**What did you do?**
I change a struct member with an immutable function of a mutable variable by using an interface:
```go
struct Int {
pub mut:
  val int
}

fn (ic mut Int) make_truth() {
  ic.val = 42
}

interface TruthMaker {
  make_truth()
}

fn call(vc TruthMaker) {
  vc.make_truth()
}

fn test_interface_with_parameter() {
  ic := Int{0} // immutable 
  call(ic)     // is changed here
  assert ic.val == 42
}
```

**What did you expect to see?**
an compiler error:
``` ic is immutable```
 
**What did you see instead?**
a valid test

### LINKED ISSUE #7338: interface bypasses struct field mutability check
**V version:** latest V
**OS:** Ubuntu 20.04

**What did you do?**
```v
struct Dog {
    pub mut:
    	name string
}

fn (mut d Dog) set_name(name string) {
	d.name = name
}

interface Animal {
    set_name(name string)   
}

fn set_animal_name(a Animal, name string) {
    a.set_name(name)   
}

fn main() {
    dog := Dog{'Doggo'}
    println(dog.name)
    set_animal_name(dog, 'Pupper')
    println(dog.name)
}
```

**What did you expect to see?**
Checker error
 
**What did you see instead?**
Compiles fine.

```
Doggo
Pupper
```

### PR DESCRIPTION
**Pull request by ntrel**
Now depends on #8097, draft until merged.
~~Fixes~~ #1081.
~~Fixes~~ #7338.

Parse mut interface method, mut interface parameter.
~~Also check method parameter mutability matches.~~ - this caused too many errors, these will need fixing in batches.

### CODE REVIEW

**Comment by ntrel** [comment]
I'm not sure why macos CI has this:

> FAIL  [20/69]     0.417 ms examples/hot_reload/bounce.v

**Comment by ntrel** [comment]
Tests green. BTW I updated the pull not to include checking receiver mutability yet when converting struct -> interface, because it caused too many errors that should be fixed in batches separately.

File: vlib/v/checker/tests/function_arg_mutable_err.out [hunk 1]
```diff
@@ -1,4 +1,4 @@
-vlib/v/checker/tests/function_arg_mutable_err.vv:1:18: error: mutable arguments are only allowed for arrays, maps, structs and pointers
+vlib/v/checker/tests/function_arg_mutable_err.vv:1:18: error: mutable arguments are only allowed for arrays, maps, aggregates and pointers
```
Comment (medvednikov) [inline vlib/v/checker/tests/function_arg_mutable_err.out:1]: V doesn't define the term aggregate anywhere, better to be explicit (structs, interfaces)

**Comment by medvednikov** [comment]
Very important fix!

File: vlib/v/checker/tests/function_arg_mutable_err.out (hunk 1 above)
Comment (ntrel) [reply vlib/v/checker/tests/function_arg_mutable_err.out]: Now done

**Pull request merged**